docs/public-api-for-1.0.md

```diff
@@ -0,0 +1,711 @@
+        public static let httpJSON: Self
+    }
+}
+```
+
+### Reducing visibility of all other existing API
+
+The existing API has biased toward a large public API. This includes all the concrete types that need chaining together
+to construct a functional OTLP backend for each signal, but also a wide API surface for extensibility.
+
+For example, the `OTelTracer` is generic over five abstractions: `OTelIDGenerator`, `OTelSampler`, `OTelPropagator`,
+`OTelSpanProcessor`, and `Clock`, with public concrete types (sometimes several) for each.
+
+Similarly, there is a public abstraction for populating attributes, `OTelResourceDetector`, with several public
+implementations and helper types: `OTelResource`, `OTelResourceDetection`, `OTelEnvironment`,
+`OTelProcessResourceDetector`, `OTelManualResourceDetector`, and `OTelEnvironmentResourceDetector`.
+
+The primary goal of Swift OTel is to provide OTLP backends for the Swift observability APIs and, where appropriate,
+support configuring the behavior according to the OTel specification.
+
+Therefore, we propose to make all the existing API not expressly discussed in the above bootstrap, backends, and
+configuration sections, internal.
+
+Reducing the API surface to just that which is required to meet these goals paves an easier path for a 1.0 release, and
+does not rule out expanding it the future to support deeper customization.
+
+Where extensibility is required, adopters should compose the opaque backends with their own logic. As a concrete
+example, if an adopter wishes to pre-sample traces in a way that is not supported in the OTel specification, they should
+construct their own wrapper type that conforms to `Tracing.Instrument`, which wraps the backend returned by
+`OTel.makeTracingBackend`.
+
+### Package structure and traits
+
+Swift OTel will provide a single library product, `OTel`, with the public API discussed above.
+
+Any other existing library products will become internal targets. Concretely, the majority of the existing `OTel`
+library product will move to an internal `OTelCore` target; and `OTLPGRPC` will be made an internal detail.
+
+Support for OTLP/gRPC and OTLP/HTTP will be gated behind package traits: `OTLPGRPC` and `OTLPHTTP`, respectively, which
+will both be enabled by default. This allows adopters to only include the dependencies for a single exporter.
+
+Swift OTel might opt to make the corresponding configuration values only reachable in code with the trait enabled, e.g.:
+
+```swift
+var config = OTel.Configuration.default
+config.traces.otlp.protocol = .grpc  // compiler error: Using the OTLP/GRPC exporter requires the `OTLPGRPC` trait enabled.
+```
+
+Attempting to use these configuration values using their corresponding environment variable values will result in a
+runtime error, e.g.:
+
+```swift
+// Assuming OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc, then...
+OTel.bootstrap()  // runtime error: Using the OTLP/GRPC exporter requires the `OTLPGRPC` trait enabled.
```

Comment:
How would this runtime error manifest? The proposed API of `OTel.bootstrap()` is marked as `throws` so I'd assume we'd throw an error in this case?